opengrok-tools/src/main/python/opengrok_tools/mirror.py

```diff
@@ -145,60 +84,38 @@ def main():
 
     logger = get_console_logger(get_class_basename(), args.loglevel)
 
+    if len(args.project) > 0 and args.all:
+        logger.fatal("Cannot use both project list and -a/--all")
+        sys.exit(1)
+
+    if not args.all and len(args.project) == 0:
+        logger.fatal("Need at least one project or --all")
+        sys.exit(1)
+
     if args.config:
         config = read_config(logger, args.config)
         if config is None:
-            logger.error("Cannot read config file from {}".format(args.config))
+            logger.fatal("Cannot read config file from {}".format(args.config))
             sys.exit(1)
     else:
         config = {}
 
-    GLOBAL_TUNABLES = [HOOKDIR_PROPERTY, PROXY_PROPERTY, LOGDIR_PROPERTY,
-                       COMMANDS_PROPERTY, PROJECTS_PROPERTY,
-                       HOOK_TIMEOUT_PROPERTY, CMD_TIMEOUT_PROPERTY]
-    diff = diff_list(config.keys(), GLOBAL_TUNABLES)
-    if diff:
-        logger.error("unknown global configuration option(s): '{}'"
-                     .format(diff))
-        sys.exit(1)
-
-    # Make sure the log directory exists.
-    logdir = config.get(LOGDIR_PROPERTY)
-    if logdir:
-        check_create_dir(logger, logdir)
-
     uri = args.uri
     if not is_web_uri(uri):
-        logger.error("Not a URI: {}".format(uri))
+        logger.fatal("Not a URI: {}".format(uri))
```

Comment:
why not use print_exc_exit ?

Reply:
I prefer logging over printing directly to `stderr`.